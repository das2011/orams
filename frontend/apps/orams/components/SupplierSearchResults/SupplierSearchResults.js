import React from 'react'
import AUpageAlert from '@gov.au/page-alerts/lib/js/react.js'
import { uniqueID } from 'shared/utils/helpers'
import styles from './SupplierSearchResults.scss'

const SupplierSearchResults = props => {
  const { searchResults, onUpdateCeilingPrice, history } = props
  const { suppliers } = searchResults || {}

  const handleUpdateCeilingPrice = supplierCode => {
    onUpdateCeilingPrice(supplierCode)
    history.push(`/edit-supplier/edit-ceiling-pricing`)
  }

  const renderTable = () => {
    if (suppliers && suppliers.length > 0) {
      return (
        <article role="main">
          <div className={styles.headingRow}>
            <div className="row">
              <div className="col-md-4 col-sm-4">Name</div>
              <div className="col-md-4 col-sm-4">Update ceiling prices</div>
              <div className="col-md-4 col-sm-4">Edit regions/service</div>
            </div>
          </div>
          {suppliers.map((supplier, id = uniqueID()) =>
            <div key={id} className={styles.supplierRow}>
              <div className="row supplier">
                <div className="col-md-4 col-sm-4 supplier-name">
                  {supplier.name}
                </div>
                <div className="col-md-4 col-sm-4 link">
                  <a
                    onClick={() => {
                      handleUpdateCeilingPrice(supplier.code)
                    }}
                  >
                    Update prices
                  </a>
                </div>
                <div className="col-md-4 col-sm-4 link">Edit services and regions</div>
              </div>
            </div>
          )}
        </article>
      )
    }

    return (
      <AUpageAlert as="info">
        <h4>There were no suppliers found for your search</h4>
      </AUpageAlert>
    )
  }

  return (
    <div className={styles.container}>
      {renderTable()}
    </div>
  )
}

export default SupplierSearchResults
