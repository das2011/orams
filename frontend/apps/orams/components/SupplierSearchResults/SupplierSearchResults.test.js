import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import expect from 'expect'
import snapshotRenderer from 'react-test-renderer'
import SupplierSearchResults from './SupplierSearchResults'

Enzyme.configure({ adapter: new Adapter() })

describe('SupplierSearchResults', () => {
  describe('when search results does not exist', () => {
    test('renders no results message', () => {
      const wrapper = shallow(<SupplierSearchResults />)

      expect(wrapper.find('AUpageAlert')).toBeTruthy()
    })
  })

  describe('when searchResults exists', () => {
    const expectedSuppliers = [
      {
        id: 1,
        name: 'Test Supplier'
      },
      {
        id: 2,
        name: 'Another Supplier'
      }
    ]
    const searchResults = { suppliers: expectedSuppliers }

    test('there are 2 supplier rows rendered', () => {
      const wrapper = shallow(<SupplierSearchResults searchResults={searchResults} />)
      const displayedSupplierRows = wrapper.find('.supplier')

      expect(displayedSupplierRows.length).toBe(2)
    })

    test('renders the names correctly', () => {
      const wrapper = snapshotRenderer.create(<SupplierSearchResults searchResults={searchResults} />).toJSON()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
