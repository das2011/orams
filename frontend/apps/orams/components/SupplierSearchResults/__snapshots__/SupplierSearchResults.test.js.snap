// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupplierSearchResults when searchResults exists renders the names correctly 1`] = `
<div
  className="container"
>
  <article
    role="main"
  >
    <div
      className="headingRow"
    >
      <div
        className="row"
      >
        <div
          className="col-md-4 col-sm-4"
        >
          Name
        </div>
        <div
          className="col-md-4 col-sm-4"
        >
          Update ceiling prices
        </div>
        <div
          className="col-md-4 col-sm-4"
        >
          Edit regions/service
        </div>
      </div>
    </div>
    <div
      className="supplierRow"
    >
      <div
        className="row supplier"
      >
        <div
          className="col-md-4 col-sm-4 supplier-name"
        >
          Test Supplier
        </div>
        <div
          className="col-md-4 col-sm-4 link"
        >
          <a
            onClick={[Function]}
          >
            Update prices
          </a>
        </div>
        <div
          className="col-md-4 col-sm-4 link"
        >
          Edit services and regions
        </div>
      </div>
    </div>
    <div
      className="supplierRow"
    >
      <div
        className="row supplier"
      >
        <div
          className="col-md-4 col-sm-4 supplier-name"
        >
          Another Supplier
        </div>
        <div
          className="col-md-4 col-sm-4 link"
        >
          <a
            onClick={[Function]}
          >
            Update prices
          </a>
        </div>
        <div
          className="col-md-4 col-sm-4 link"
        >
          Edit services and regions
        </div>
      </div>
    </div>
  </article>
</div>
`;
